### PR TITLE
Working on BootNOMP

### PR DESCRIPTION
**Pull request by mooleshacat**
Hey I am going to work on BootNOMP (already have done quite a bit)

Will submit merges shortly. Sorry no other way to message you.

Probably should enable issues tracker so people can contact for issues...

Talk soon.

admin@pizone.tk

### CODE REVIEW

**Comment by 1301313Y** [comment]
Oh thank you for pointing that out, I had completely ignored that for some reason. Will open it up right now! Thank you for contributing as well, can't wait to see your work! 😄

**Comment by leshacat** [comment]
Yes I am working on it for myself, my community pool, and to help the project... The block explorer is almost done, you will like the changes I promise, and if not just reject it :p

I also going to make lots of coins/coinname.json files and pool_config/coinname_example.json files.

Almost ready for a pull... Just need to fix a bunch more things :d 

You can preview the pool / block explorer @ http://mn0.pizone.tk:44420/ but currently has a bug and sometimes does not show the blocks (refresh a few times every 10 mins it shows eventually)

I have my miner on MOG right now :)

**Comment by leshacat** [comment]
**I am not sure if you want to merge now, or wait for me to finish block explorer page...**

I have changed some alignment here and there, added a donation address feature, modified block explorer (not finished yet, bug), and coin/pool_config files and examples for MogwaiCoin (MOG), SecureTag (STG) and RavenCoin (RVN). I plan to add more as I add them to my pool.

I will try not to push unless there are no confirmed bugs. Still getting used to git.

If you do merge, be sure to carefully inspect the changes.

If you prefer, you can wait until I complete the block explorer and a few more modifications (sitename/logo in config file, etc.)

I know you are busy and this is not my personal request forum, but would you be able to work on (at some point) a replacement or additional options to redis?

IMHO redis is a memory cache utility, not a database, and I have been having memory issues (lol) on my VPS where lots of people host pools. I am thinking MySQL, MariaDB, and/or MongoDB.

Let me know ;)

File: website/key.html [hunk 1]
```diff
@@ -138,7 +138,6 @@ <h4>Mining key generation or input options:</h4>
     <div class="roundedWrapper">
         <div class="heading">Coin formatted keys</div>
         <select id="coinSelect">
-            <option value="">Select a coin</option>
```
Comment (leshacat) [inline website/key.html]: This may be a mistake... Maybe need to keep this line...?

File: website/key.html [hunk 2]
```diff
@@ -244,7 +243,7 @@ <h4>Mining key generation or input options:</h4>
     var printFrame = $id('printFrame');
     var printFrameDoc = printFrame.contentDocument || printFrame.contentWindow.document;
     printFrameDoc.body.innerHTML = '';
-    printFrameDoc.write('Private key hex data:<br> ' + wallet.getPrivateKeyHex() + '<script> function printPage(){ print() } <\/script>');
+    printFrameDoc.write('Private key hex data:<br /> ' + wallet.getPrivateKeyHex() + '<script> function printPage(){ print() } <\/script>');
```
Comment (leshacat) [inline website/key.html]: Part of my pathetic attempt to W3 validate page with https://validator.w3.org/ while debugging

File: website/pages/history.html [hunk 3]
```diff
@@ -76,7 +76,7 @@
   </div>
   <h4 class="display-4 ml-3 mt-0 mb-3" style="font-size: 24px;">Last 50 Confirmed Blocks</h4>
   <fieldset>
-  <form action="" class="form-group" method="post">
+  <form action="#" class="form-group" method="post">
```
Comment (leshacat) [inline website/pages/history.html]: W3 (must have an action=)

File: libs/paymentProcessor.js [hunk 4]
```diff
@@ -158,7 +158,7 @@ function SetupForPool(poolOptions, setupFinished) {
         daemon.cmd('getnetworkinfo', params,
           function(result) {
             if (!result || result.error || result[0].error || !result[0].response) {
-              logger.error('Error with RPC call getnetworkinfo ' + JSON.stringify(result[0].error));
+              logger.error('@@@!WARN:4!@@@ Error with RPC call getnetworkinfo ' + JSON.stringify(result[0].error));
```
Comment (leshacat) [inline libs/paymentProcessor.js]: I will take these out... Was debugging.

File: scripts/cli.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 var net = require('net');
 
-var defaultPort = 17117;
+var defaultPort = 17000;
```
Comment (leshacat) [inline scripts/cli.js]: Need info in README.md regarding changing this setting for blocknotify to work

File: website/pages/blocks.html [hunk 6]
```diff
@@ -32,66 +33,87 @@
...
   <div class="col-md-5 table-responsive">
-    <table class="table table-bordered">
+    <table class="table table-bordered" style="width: 550px;">
       <thead>
         <tr>
-          <th><i class="fa fa-history"></i> Block History For</th>
-          <th>
-            <select class="form-control" id="coinSelect">
-              <option class="btn btn-primary dropdown-toggle" value="{{=sortedPools[0]}}" type="button" id="coinDropDown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-                {{=sortedPools[0].charAt(0).toUpperCase() + sortedPools[0].slice(1)}}
```
Comment (leshacat) [inline website/pages/blocks.html]: This whole file needs work before merge (bugs, not done yet)

**Review by mooleshacat — COMMENTED** [review]
This is showing both of ours merged, including your recent changes is it? I deleted a few things that I did not think I coded wasn't sure how they got in there...

Anyways I have looked over the entire merge, the only page needing work it looks is blocks.html which I am working on.

Some of the "NotoHash" I've replaced with EasyX but I will be making it a config option to set text or a logo.

I hope I didn't screw it up and you can deny the changes if I did delete some file (package.json.lock?)

I am going to revamp the debug logging system after the website changes, add in a sort of "stack trace" so we can tell what function is throwing the error. Also I will do some documentation - I am really good at it :+1:

**Comment by leshacat** [comment]
block_explorer
That is the block explorer. I will be getting icons for all the coins too, so they are included with the coin configs and examples.

The explorer will show all transactions mixed, or option to show one single coins payments.

It will eventually show immature blocks and confirmation numbers.

Of to work on that now :)

**Comment by leshacat** [comment]
Block explorer is done, just need to add in immature blocks, and confirmations :)

**Comment by SkinnyPeteTheGiraffe** [comment]
Just reviewed everything and it seems great, thank you for contributing! 👍

**Comment by mooleshacat** [comment]
No problem. I was a bit worried :P Just working on block explorer to show immature blocks and their confirmations.

I had issues with my pool I was sorting out. RVN was not working - had wrong walletd version.

Back to work on block explorer!

Will make new pull request when done block explorer and a few other options.

**Pull request merged**